### PR TITLE
[PING][TRACERT] Fully support Win64 ICMP_ECHO_REPLY32

### PR DESCRIPTION
**Pull request by SergeGautherie**

### CODE REVIEW

**Comment by ThFabba** [comment]
What's the context for this? The documentation makes no mention of the 32 bit version of the structure, there's no explanation in the commit why this is neccessary....

In short... what does this do?

**Comment by SergeGautherie** [comment]
> context

tracert already uses this structure to calculate the reply size, (so I assumed this PR is obvious)
ping even accesses `Options` field.

> documentation makes no mention

Which documentation?
MS ones do. For example:
https://docs.microsoft.com/en-us/windows/win32/api/ipexport/ns-ipexport-icmp_echo_reply
'On a 64-bit platform, the ICMP_ECHO_REPLY32 structure should be used.'

> what does this do?

ReactOS already has this structure. `POINTER_32` is needed on Win64.

In short... It forces 32-bit structures on 64-bit OS.

NB:
I am working on tests and code too...

**Comment by ThFabba** [comment]
Doing it either way works fine, at least on Win10 (it seems to determine the right structure based on size). And of course this change will break ROS x64, because we simply use the 64 bit structure in iphlpapi. Hence I was wondering what this actually fixed.

The documentation link does support this change, I hadn't looked at that particular page before. Guess we'll have to fix iphlpapi some day then...

**Comment by SergeGautherie** [comment]
> at least on Win10

I don't know, I trust you. I assume we are interested on XP/S2003 x64.

> break ROS x64

I concur iphlpapi will need fixing too, but that is separate/future.

Until that is further investigated/implemented, we could use `#if !defined(__REACTOS__) && defined(_WIN64)` instead, if you think that is worth it.

> what this actually fixed

tracert code was inconsistent, so (already) buggy (on ROS x64).
As for ping, the main point is to sync both codes...

**Comment by ThFabba** [comment]
> I concur iphlpapi will need fixing too, but that is separate/future.

It is not separate, and that's kind of the point.

What happened here is that we introduced a regression (x64 ping will no longer work on ROS, it may have worked before). "Make the code consistent", is _very_, _very_ far from a reasonable justification for doing that.
And this is exactly why I ask silly questions like "what does this fix" -- because every change has a risk of regression. That's even more true if you don't actually test the change -- which in this case would have been impossible because networking does not work on x64 yet. Touching code that can't be tested should probably be something we expressly forbid, but at the very least should require strong justification.

I'm okay with leaving this change in master, since it clearly follows the documentation, and x64 networking (including iphlpapi) will require more work anyway. But this is not a pattern to keep following.
